enterprise/ha/src/main/java/org/neo4j/kernel/impl/ha/ClusterManager.java

```diff
@@ -231,6 +248,41 @@ public static Provider clusterOfSize( int memberCount )
     }
 
     /**
+     * Find an available port number which is also not included in the except list.
+     * @param minPort minimum port number to probe
+     * @param maxPort maximum port number to probe
+     * @param except port numbers which should be considered unavailable and already taken
+     * @return a port number
+     * @throws IOException if no open port could be found
+     */
+    private static int findFreePort( final int minPort, final int maxPort, final Collection<Integer> except )
+            throws IOException
```

Comment:
How come? Tests are not performance critical so I'd always go with built-in types instead of our home grown NIH types.